pkg/devfile/devfile.go

```diff
@@ -250,3 +253,30 @@ func DownloadDevfileAndDockerfile(url string) ([]byte, string, []byte) {
 func ScanRepo(log logr.Logger, a Alizer, localpath string, devfileRegistryURL string, source appstudiov1alpha1.GitSource) (map[string][]byte, map[string]string, map[string]string, error) {
 	return search(log, a, localpath, devfileRegistryURL, source)
 }
+
+// UpdateLocalDockerfileURItoAbsolute takes in a Devfile, and a DockefileURL, and returns back a Devfile with any local URIs to the Dockerfile updates to be absolute
+func UpdateLocalDockerfileURItoAbsolute(devfile data.DevfileData, dockerfileURL string) (data.DevfileData, error) {
+	devfileComponents, err := devfile.GetComponents(common.DevfileOptions{ComponentOptions: common.ComponentOptions{
+		ComponentType: v1alpha2.ImageComponentType,
+	}})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, comp := range devfileComponents {
+		if comp.Image != nil && comp.Image.Dockerfile != nil {
+			// If the DockerfileURI is local (i.e. not a fully qualified URL) overwrite it to be the given Dockerfile URL, otherwise ignore it
```

Comment:
Why not? If Component Source has DockerfileURL, we probably want to overwrite it always?

Reply:
My thinking was that we wouldn't want to overwrite it, but there be cases where users pass in DockerfileURL to overwrite a fully qualified URL in the devfile that isn't resolvable or maybe private. I'll update it

Reply:
Updated.